### ISSUE TITLE
[css-break] Empty fragment at fragmentainer boundary

### ISSUE THREAD

**Issue by mstensho** [w3c/csswg-drafts#1529]
https://drafts.csswg.org/css-break/

If we have an empty, zero-height block, and it occurs exactly at a fragmentainer boundary, where should it end up? In the former fragmentainer, or in the latter?

<div style="columns:2: column-fill:auto; height:100px;">
    <div style="height:100px;"></div>
    <div id="empty"></div>
    <div style="height:50px;"></div>
</div>

Is #empty in the first or second column? If the answer is "second", then how about:

<div style="columns:2: column-fill:auto; height:100px;">
    <div style="height:100px;"></div>
    <div id="empty"></div>
</div>

Should we establish a second column just to hold an empty fragment?

This may not matter for painting and hit-testing, but it's exposable via e.g. getClientRect().

**Comment by MatsPalmgren** [comment]
Definitely in the former fragment. It's very observable if the zero-height block has abs.pos. children.

(BTW, your example markup isn't visible.)

**Comment by mstensho** [comment]
(Sorry - fixed the markup)

**Comment by fantasai** [comment]
This is technically undefined, since we don't define between which elements one is supposed to break--just where one is allowed / discouraged / forbidden from breaking. I'd imagine it'd go in the earlier fragmentainer, since that's what typically happens in other fragmenting modes like inline line breaking and flex line breaking.

**Comment by SelenIT** [comment]
I'd prefer it going in the earlier fragmentainer, too. I expect the new fragmentainer to be created only if the former one is overflowed, and adding a zero-sized item shouldn't cause overflowing if the fragmentainer wasn't overflowed without it.

**Comment by tabatkins** [comment]
Yes, everywhere else we do *explicitly* define breaking, we mandate that zero-size things at a boundary live on the previous line, not the next. (See flexbox, for example.)

Making this clearer in Fragmentation would be good.

**Comment by fantasai** [comment]
Agenda+ to ask the WG about mandating, specifically for zero-sized elements, that they stay with the earlier fragmentainer is a good idea.

**Comment by fantasai** [comment]
OK, added the following:

> Additionally, CSS imposes one requirement: a zero-sized box fragment, since it does not take up space, must appear on the earlier side of a fragmentation break if it is able to fit within the fragmentainer.
>
> Note: A zero-sized box fragment can be pushed to the next fragmentainer if it is placed immediately after content that itself overflows the fragmentainer.

Agenda+ to check on the subtleties of a zero-sized fragment that's placed after an overflowing piece of content. :/

**Comment by tabatkins** [comment]
The most reasonable answer seems clear, and is already reflected [in Flexbox](https://drafts.csswg.org/css-flexbox/#algo-line-break): they go to the next fragmentainer.
